docs/src/tutorial.md

```diff
@@ -0,0 +1,157 @@
+be achieved by taking the intersection of both results. For example, let us consider the function ``g(x) = x^2 - 2x + 1``, which we want to evaluate over the domain ``X_g = [0, 4]``. Let us first try using plain interval arithmetic
+
+```@example tutorial
+g(x) = x^2 - 2*x + 1
+Xg = 0..4
+enclose(g, Xg, NaturalEnclosure()) # this is equivalent to enclose(g, Xg)
+```
+
+Now let us bound the range using [`MeanValueEnclosure`](@ref), which uses the mean-value form of the function.
+
+```@example tutorial
+enclose(g, Xg, MeanValueEnclosure())
+```
+
+As you can see, there is no clear winner and a better enclosure could be obtained by taking the intersection of the two results. This can be easily done in one pass by passing to `enclose` a vector of methods
+
+```@example tutorial
+enclose(g, Xg, [NaturalEnclosure(), MeanValueEnclosure()])
+```
+
+### Using algorithms from external libraries
+
+Not all algorithms are implemented in this package, some are from external libraries. To keep the start-up time of RangeEnclosures.jl reasonable, routines from external packages
+are not imported by default and if you want to use those, you will need to also import the required package. For example, suppose you want to bound the previous function using the Moore-Skelboe algorithm. Trying the following will fail
+
+```julia
+julia> enclose(f, X, MooreSkelboeEnclosure())
+```
+
+```
+ERROR: AssertionError: package 'IntervalOptimisation' not loaded (it is required for executing `enclose`)
+...
+```
+
+This is because the algorithm is implemented in [`IntervalOptimisation.jl`](https://github.com/juliaintervals/intervaloptimisation.jl) and to use it you need to import it first (note that you need to have it installed before importing it). Let us fix our example
+
+```@example tutorial
+using IntervalOptimisation
+enclose(f, X, MooreSkelboeEnclosure())
+```
+
+### Bounding multivariate functions
+
+While our previous examples were in 1D, the techniques generalize to multivariate functions ``\mathbb{R}^n\rightarrow\mathbb{R}``, the only difference is that the domain, instead of being an interval, should be an `IntervalBox`. For example, if we want to bound the function
+
+```math
+h(x_1, x_2) = \sin(x_1) - cos(x_2) - \sin(x_1)\cos(x_1)
+```
+
+over the domain ``X_h = [-5, 5] \times [-5, 5]``. It can be done as follows
+
+```@example tutorial
+h(x) = sin(x[1]) - cos(x[2]) - sin(x[1]) * cos(x[1])
+Xh = IntervalBox(-5..5, 2) # note this is an interval box and not an interval
+Yh = enclose(h, Xh, BranchAndBoundEnclosure())
+```
+
+we can visualize the result for example with the following script
+
+```@example tutorial
+x = y = -5:0.1:5
+f(x, y) = h([x, y])
+surface(x, y, [sup(Yh) for _ in x, _ in y], α=0.4, legend=:none, size=(500, 500))
+surface!(x, y, [inf(Yh) for _ in x, _ in y], α=0.4)
+surface!(x, y, f.(x', y), zlims=(-4, 4))
+savefig("ex2.png") # hide
```

Comment:
You do not have to put these `savefig`s to show plots in the online documentation.

Reply:
really? It saves automatically?

Reply:
Yes, it shows the last result of that block, which in this case is a plot.

Reply:
cool, I'll update

Reply:
If the plots become too large, we can use `DisplayAs`, but I think in this package we will not run into trouble.